### PR TITLE
Implement  a Zipkin exporter

### LINKED ISSUE #18: Implement a zipkin exporter

### PR DESCRIPTION
**Pull request by nacho-undefinedlabs**
Also added some imporvemtns to Api and Sdk that were obvious while developing this exporter

Close #18

### CODE REVIEW

File: Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            processResources(state: &attributeEnumerationState, key: $0.key, value: $0.value)
+        }
+
+        var localEndpoint = defaultLocalEndpoint
+
+        var serviceName = attributeEnumerationState.serviceName
+
+        if serviceName != nil, !serviceName!.isEmpty {
+            if let serviceNamespace = attributeEnumerationState.serviceNamespace, !serviceNamespace.isEmpty {
+                serviceName = serviceNamespace + "." + serviceName!
```
Comment (SergeyKanzhelev) [inline Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift:61]: I think this logic was implemented in places, but spec forbid to combine namespace and name

File: Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    private static func toSpanKind(otelSpan: SpanData) -> String {
+        switch otelSpan.kind {
+        case .server:
+            return "SERVER"
+        case .producer:
+            return "PRODUCER"
+        case .consumer:
+            return "CONSUMER"
+        default:
+            return "CLIENT"
```
Comment (SergeyKanzhelev) [inline Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift:125]: I think default is empty string or something. Please refer to the spec

File: Sources/Exporters/Zipkin/Implementation/ZipkinSpan.swift [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+                object["timestamp"] = $0.timestamp
+                object["value"] = $0.value
+                return object
+            }
+
+            output["annotations"] = annotationsArray
+        }
+
+        if tags.count > 0 {
+//            let tagsArray: [Any] = tags.map {
```
Comment (SergeyKanzhelev) [inline Sources/Exporters/Zipkin/Implementation/ZipkinSpan.swift:83]: do you want to keep commented lines?

File: Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift (hunk 1 above)
Comment (nacho-undefinedlabs) [reply Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift:61]: True, fixed and added the namespace as a Zipkin tag as the spec says.

File: Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift (hunk 2 above)
Comment (nacho-undefinedlabs) [reply Sources/Exporters/Zipkin/Implementation/ZipkinConversionExtension.swift:125]: True, should not be set

File: Sources/Exporters/Zipkin/Implementation/ZipkinSpan.swift (hunk 3 above)
Comment (nacho-undefinedlabs) [reply Sources/Exporters/Zipkin/Implementation/ZipkinSpan.swift:83]: Nope, thanks for catching

**Pull request merged**